### PR TITLE
Internationalisation UI for snippets

### PR DESCRIPTION
**Pull request by kaedroho**
This PR adds a locale selector to the snippet listing, edit, and create views

It looks like this
image

### CODE REVIEW

File: wagtail/snippets/static_src/wagtailsnippets/js/snippet-chooser-modal.js [hunk 1]
```diff
@@ -13,13 +13,24 @@ SNIPPET_CHOOSER_MODAL_ONLOAD_HANDLERS = {
             });
         }
 
-        var searchUrl = $('form.snippet-search', modal.body).attr('action');
+        var searchForm$ = $('form.snippet-search', modal.body);
```
Comment (zerolab) [inline wagtail/snippets/static_src/wagtailsnippets/js/snippet-chooser-modal.js:16]: note that the convetion for jQuery vars is `$var` rather than `var$`

Comment (kaedroho) [reply wagtail/snippets/static_src/wagtailsnippets/js/snippet-chooser-modal.js:16]: Oops, it's been a while. Thanks!

Comment (kaedroho) [reply wagtail/snippets/static_src/wagtailsnippets/js/snippet-chooser-modal.js]: I've moved this part out of this PR so the rest can be reviewed.

File: wagtail/snippets/templates/wagtailsnippets/snippets/create.html [hunk 2]
```diff
@@ -2,10 +2,31 @@
 {% load i18n wagtailadmin_tags %}
 {% block titletag %}{% blocktrans with snippet_type_name=model_opts.verbose_name %}New  {{ snippet_type_name }}{% endblocktrans %}{% endblock %}
 {% block content %}
-    {% trans "New" as new_str %}
-    {% include "wagtailadmin/shared/header.html" with title=new_str subtitle=model_opts.verbose_name icon="snippet" tabbed=1 merged=1 %}
+    <header class="merged tab-merged">
+        <div class="row">
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/snippets/templates/wagtailsnippets/snippets/create.html:6]: Maybe we could the locale filter into the shared header, in a similar way we have search there? I've implemented this to use the exact same context variables that the page selector uses, and I imagine any future translatable content types (settings, modeladmin) would use the same too

**Review by gasman — APPROVED** [review]
Happy with the functionality here, so let's go ahead and merge and worry about refactoring the header separately :-)

**Comment by kaedroho** [comment]
Thanks!

**Pull request merged**